plugins/inputs/wireless/wireless_mac.go

```diff
@@ -0,0 +1,56 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+)
+
+// default executable path & flags
+const (
+	OSXCMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport"
+	FLAG   = "-I"
+)
+
+func (w *Wireless) Gather(acc telegraf.Accumulator) error {
+	cmd := exec.Command(OSXCMD, FLAG)
+	wireless, err := internal.StdOutputTimeout(cmd, 2*time.Second) // cmd.Output()
+	if err != nil {
+		return err
+	}
+	metrics, tags := w.loadMacWirelessTable(wireless)
+	acc.AddFields("wireless", metrics, tags)
+	return nil
+}
+
+func (w *Wireless) loadMacWirelessTable(table []byte) (map[string]interface{}, map[string]string) {
```

Comment:
Maybe `parseAirportOutput`? I'm also OK with keeping the name but the `load` part is a bit confusing...

Reply:
It is called `loadWirelessTable` in the `_linux` version. I was trying to maintain consistency.

Reply:
Ok, then please keep the name!

Reply:
On the other hand, the linux version loads info from `/proc/net/wireless` so that name makes sense. Here we are loading/parsing output of command `airport`.